### ISSUE TITLE
Unexpected `Lint/ImplicitStringConcatenation` offense with `prism` parser

### ISSUE THREAD

**Issue by viralpraxis** [rubocop/rubocop#13625]
## Expected behavior

```ruby
# test.rb

<<_
  123\n456\n789
_
```

I expect `rubocop test.rb --only Lint/ImplicitStringConcatenation` with both `parser_prism` and `parser_whitequark` to produce same output.

## Actual behavior

parser_whitequark:

```console
rubocop test.rb --only Lint/ImplicitStringConcatenation

Inspecting 1 file
.

1 file inspected, no offenses detected
```

parser_prism:

```console
rubocop test.rb --only Lint/ImplicitStringConcatenation

Offenses:

test.rb:3:1: W: [Correctable] Lint/ImplicitStringConcatenation: Combine  and  into a single string literal, rather than using implicit string concatenation.
_


1 file inspected, 1 offense detected, 1 offense autocorrectable
```

AST:

```ruby
RuboCop::AST::ProcessedSource.from_file("test.rb", 3.4, parser_engine: :parser_whitequark).ast
=> s(:str, "  123\n456\n789\n")

RuboCop::AST::ProcessedSource.from_file("test.rb", 3.4, parser_engine: :parser_prism).ast
=> 
s(:dstr,
  s(:str, "  123\n"),
  s(:str, "456\n"),
  s(:str, "789\n"))
```
## Steps to reproduce the problem

See above.

## RuboCop version

```
1.69.2 (using Parser 3.3.6.0, rubocop-ast 1.37.0, analyzing as Ruby 3.3, running on ruby 3.3.6) [x86_64-linux]
1.69.2 (using Prism 1.2.0, rubocop-ast 1.37.0, analyzing as Ruby 3.3, running on ruby 3.3.6) [x86_64-linux]
```

**Comment by Earlopain** [comment]
This is a duplicate of #12878. I currently have PRs open against prism to work towards fixing this.

**Comment by viralpraxis** [comment]
@Earlopain got it, thanks
